### PR TITLE
Csu merge

### PR DESCRIPTION
**Pull request by brooksdavis**
Replacing POSITION_INDEPENDENT_STARTUP with PIC and removing the additions the PICFLAG seems to build for mips.

### CODE REVIEW

File: lib/csu/riscv64c/crt1_c.c [hunk 1]
```diff
@@ -114,7 +114,7 @@ _start(void *auxv,
 	}
 
 	/* For -pie executables rtld will initialize the __cap_relocs */
-#ifndef POSITION_INDEPENDENT_STARTUP
+#ifndef PIC
```
Comment (arichardson) [inline lib/csu/riscv64c/crt1_c.c:117]: As far as I recall, we originally needed the PIC define even for static MIPS. However, this is no longer the case so we should use the same macro as upstream.

File: lib/csu/riscv64c/Makefile [hunk 2]
```diff
@@ -2,34 +2,6 @@
...
-
-crt1.o:	crt1_c.o crtbrand.o ignore_init_note.o
-	${LD} ${_LDFLAGS} -o crt1.o -r crtbrand.o ignore_init_note.o crt1_c.o
-
-Scrt1_c.o: crt1_c.c
-	${CC} ${CFLAGS} ${PICFLAG} -c -o ${.TARGET} ${.CURDIR}/crt1_c.c
-
-Scrt1.o: Scrt1_c.o crtbrand.o ignore_init_note.o
-	${LD} ${_LDFLAGS} -o Scrt1.o -r crtbrand.o ignore_init_note.o Scrt1_c.o
+CFLAGS_CRTS+=	-DPOSITION_INDEPENDENT_STARTUP=1
```
Comment (arichardson) [inline lib/csu/riscv64c/Makefile:5]: Probably not needed anymore?

Comment (brooksdavis) [reply lib/csu/riscv64c/Makefile:5]: oops, delete the MIPS one and failed to delete this one.

File: lib/csu/Makefile.inc [hunk 3]
```diff
@@ -8,15 +8,62 @@ NO_WMISSING_VARIABLE_DECLARATIONS=
...
 
 CFLAGS+=	-fno-asynchronous-unwind-tables
 CFLAGS+=	-fno-omit-frame-pointer
+CFLAGS+=	-I${.CURDIR:H}/common \
+		-I${SRCTOP}/lib/libc/include
+
+.if ${MACHINE_ABI:Mpurecap}
+# Purecap code requires crt_init_globals to be called before using jump tables
+CFLAGS+=	-fno-jump-tables
+CFLAGS+=	-DCRT_IRELOC_SUPPRESS
```
Comment (bsdjhb) [inline lib/csu/Makefile.inc:31]: I had left this one in each file, but it's fine to do here.

**Review by bsdjhb — APPROVED** [review]
I was about to push a similar change for review, looks good to me.

File: lib/csu/Makefile.inc (hunk 3 above)
Comment (brooksdavis) [reply lib/csu/Makefile.inc:31]: With just riscv and mips I'd have kept it there, but with morello on the way and also using these flags it felt like a little clutter here was worth it.

**Pull request merged**